tests/pipelines/test_train_step.py

```diff
@@ -54,45 +56,91 @@ def set_up_train_step(pipeline_root: Path):
     transformed_dataset.to_parquet(str(split_step_output_dir / "validation.parquet"))
     transformed_dataset.to_parquet(str(split_step_output_dir / "train.parquet"))
 
+    return train_step_output_dir
+
+
+# Sets up the constructed TrainStep instance
+def setup_train_step(pipeline_root: Path, use_tuning: bool):
     pipeline_yaml = pipeline_root.joinpath(_PIPELINE_CONFIG_FILE_NAME)
-    pipeline_yaml.write_text(
-        """
-        template: "regression/v1"
-        target_col: "y"
-        profile: "test_profile"
-        run_args:
-            step: "train"
-        experiment:
-          name: "demo"
-          tracking_uri: {tracking_uri}
-        steps:
-          train:
-            using: estimator_spec
-            estimator_method: sklearn.linear_model.SGDRegressor
-            tuning:
-                enabled: false
-        """.format(
-            tracking_uri=mlflow.get_tracking_uri()
+    if use_tuning:
+        pipeline_yaml.write_text(
+            """
+            template: "regression/v1"
+            target_col: "y"
+            profile: "test_profile"
+            run_args:
+                step: "train"
+            experiment:
+                name: "demo"
+                tracking_uri: {tracking_uri}
+            steps:
+                train:
+                    using: estimator_spec
+                    estimator_method: tests.pipelines.test_train_step.estimator_fn
+                    estimator_params:
+                        alpha: 0.1
+                        penalty: l1
+                    tuning:
+                        enabled: true
+                        max_trials: 2
+                        sample_fraction: 0.5
+                        parameters:
+                            alpha:
+                                distribution: "uniform"
```

Comment:
Can we add test coverage to make sure that other distributions are supported too?